pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -160,12 +161,22 @@ func (r *AtlasClusterReconciler) Delete(e event.DeleteEvent) error {
 		return fmt.Errorf("cannot build Atlas client: %w", err)
 	}
 
-	_, err = atlasClient.Clusters.Delete(context.Background(), project.Status.ID, cluster.Spec.Name)
-	if err != nil {
-		return fmt.Errorf("cannot delete Atlas cluster: %w", err)
-	}
+	go func() {
+		timeout := time.Now().Add(workflow.DefaultTimeout)
+
+		for time.Now().Before(timeout) {
+			_, err = atlasClient.Clusters.Delete(context.Background(), project.Status.ID, cluster.Spec.Name)
+			if err != nil {
```

Comment:
what do you think if we check here the error that is returned in case the cluster is already removed - and break the loop in this case?

Reply:
Good catch!